src/annotator/anchoring/pdf.js

```diff
@@ -41,11 +41,35 @@ function getNodeTextLayer(node) {
 /**
  * Returns the view into which a PDF page is drawn.
  *
+ * If called while the PDF document is still loading, this will delay until
+ * the document loading has progressed far enough for a `PDFPageView` and its
+ * associated `PDFPage` to be ready.
+ *
  * @param {number} pageIndex
- * @return {PDFPageView}
+ * @return {Promise<PDFPageView>}
  */
-function getPage(pageIndex) {
-  return PDFViewerApplication.pdfViewer.getPageView(pageIndex);
+async function getPageView(pageIndex) {
+  const pdfViewer = PDFViewerApplication.pdfViewer;
+  let pageView = pdfViewer.getPageView(pageIndex);
+
+  if (!pageView || !pageView.pdfPage) {
+    // If the document is still loading, wait for the `pagesloaded` event.
+    //
+    // Note that loading happens in several stages. Initially the page view
+    // objects do not exist (`pageView` will be nullish), then after the
+    // "pagesinit" event, the page view exists but it does not have a `pdfPage`
+    // property set, then finally after the "pagesloaded" event, it will have
+    // a "pdfPage" property.
+    pageView = await new Promise(resolve => {
+      const onPagesLoaded = () => {
+        document.removeEventListener('pagesloaded', onPagesLoaded);
+        resolve(pdfViewer.getPageView(pageIndex));
+      };
+      document.addEventListener('pagesloaded', onPagesLoaded);
```

Comment:
if this timeouts or breaks, should we bother with error handling or a reject case?

Reply:
If an exception occurs in this callback, the promise will reject. The document loading process could take a long time, and I'm not sure whether it depends on purely local computation or whether it can include network fetches. On this basis, any timeout would need to be generous I think.